test/components/along_test.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:test/test.dart';
+import 'package:turf/along.dart';
+import 'package:turf/distance.dart';
+import 'package:turf/helpers.dart';
+import 'package:turf/length.dart';
+
+void main() {
+  test('along - negative distance should count backwards', () {
+    final viaToEndDistance =
+        distance(Point(coordinates: via), Point(coordinates: end), Unit.meters);
+    expect(viaToEndDistance.round(), equals(198));
+    final resolvedStartPoint = along(line, -1 * viaToEndDistance, Unit.meters);
+    expect(resolvedStartPoint.coordinates, equals(start));
+  });
+  test('along - to start point', () {
+    final resolvedStartPoint = along(line, 0, Unit.meters);
+    expect(resolvedStartPoint.coordinates, equals(start));
+  });
+  test('along - to point between start and via', () {
+    final startToViaDistance = distance(
+        Point(coordinates: start), Point(coordinates: via), Unit.meters);
+    expect(startToViaDistance.round(), equals(57));
+    final resolvedViaPoint = along(line, startToViaDistance / 2, Unit.meters);
+    expect(resolvedViaPoint.coordinates.lat.toStringAsFixed(6),
+        equals('55.709028'));
+    expect(resolvedViaPoint.coordinates.lng.toStringAsFixed(6),
+        equals('13.185096'));
+  });
+  test('along - to via point', () {
+    final startToViaDistance = distance(
+        Point(coordinates: start), Point(coordinates: via), Unit.meters);
+    expect(startToViaDistance.round(), equals(57));
+    final resolvedViaPoint = along(line, startToViaDistance, Unit.meters);
+    expect(resolvedViaPoint.coordinates, equals(via));
+  });
+  test('along - to point between via and end', () {
+    final startToViaDistance = distance(
+        Point(coordinates: start), Point(coordinates: via), Unit.meters);
+    final viaToEndDistance =
+        distance(Point(coordinates: via), Point(coordinates: end), Unit.meters);
+    expect(startToViaDistance.round(), equals(57));
+    expect(viaToEndDistance.round(), equals(198));
+    final resolvedViaPoint =
+        along(line, startToViaDistance + viaToEndDistance / 2, Unit.meters);
+    expect(resolvedViaPoint.coordinates.lat.toStringAsFixed(6),
+        equals('55.708330'));
+    expect(resolvedViaPoint.coordinates.lng.toStringAsFixed(6),
+        equals('13.186555'));
+  });
+  test('along - to end point', () {
+    final len = length(line, Unit.meters);
+    expect(len.round(), equals(254));
+    final resolvedEndPoint = along(line, len, Unit.meters);
+    expect(resolvedEndPoint.coordinates, equals(end));
+  });
+  test('along - beyond end point', () {
+    final len = length(line, Unit.meters);
+    expect(len.round(), equals(254));
+    final resolvedEndPoint = along(line, len + 100, Unit.meters);
+    expect(resolvedEndPoint.coordinates, equals(end));
+  });
+}
+
```

Comment:
I really liked your tests, but it would be great if you could a very simple test case: 
- along call where no unit is specified to state that the default unit is kilometer and the unit parameter is used at all and is not hardcoded to meters.

Simply copying an existing test (such as 'along - to end point') would be perfectly fine.

Reply:
Thanks for the feedback. I duplicated one test and changed to use default unit.